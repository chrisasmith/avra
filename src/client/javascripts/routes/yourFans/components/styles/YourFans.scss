@import '../../../../../stylesheets/modules/gridPage.scss';
