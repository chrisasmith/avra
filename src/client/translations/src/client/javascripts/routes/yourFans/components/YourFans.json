[
  {
    "id": "your_fans.title",
    "defaultMessage": "Your Fans"
  }
]