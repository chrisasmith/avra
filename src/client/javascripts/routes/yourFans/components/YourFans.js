import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { FormattedMessage } from 'react-intl'
import * as yourFansActions from '../../../reducers/pages/yourFansActions'
import FilterBar from '../../../components/Filter/FilterBar'
import { createLoadingSelector, createErrorMessageSelector } from '../../../reducers/selectors'
import styles from './styles/YourFans.scss'
import Tabs from './Tabs'
import DASHBOARD_DATA from './Data'

const propTypes = {
  getHomeData: PropTypes.func.isRequired,
  homePageDataSuccess: PropTypes.func.isRequired,
  location: PropTypes.object.isRequired,
  filter: PropTypes.object.isRequired,
  children: PropTypes.element,
}

export const PROPERTY_TYPE = {
  SPORT: 'sport',
  LEAGUE: 'league',
  TEAM: 'team',
}

class YourFansPage extends Component {
  constructor(props, context) {
    super(props, context)

    this.onFilterChange = this.onFilterChange.bind(this)

    this.state = {
      filter: { ...props.filter, ages: { ...props.filter.ages } },
    }
  }

  componentDidMount() {
    this.props.homePageDataSuccess(DASHBOARD_DATA)

    this.props.getHomeData()
  }

  onFilterChange(filter) {
    this.setState({ filter }, () => this.props.getHomeData(filter))
  }

  render() {
    return (
      <div className={styles.columns}>
        <div>
          <FilterBar
            onFilterChange={this.onFilterChange}
          />
        </div>
        <div>
          <div className={styles.title}>
            <FormattedMessage
              id="your_fans.title"
              defaultMessage="Your Fans"
            />
          </div>
          <div className={styles.tabs}><Tabs location={this.props.location} /></div>
          {this.props.children}
        </div>
      </div>
    )
  }
}

function mapStateToProps(state) {
  const loadingSelector = createLoadingSelector([yourFansActions.GET_HOME_PAGE.SELF])
  const errorSelector = createErrorMessageSelector([yourFansActions.GET_HOME_PAGE.SELF])
  const filter = state.filters.filter
  return { ...state, filter, isFetching: loadingSelector(state), errorMessage: errorSelector(state) }
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ ...yourFansActions }, dispatch)
}

YourFansPage.propTypes = propTypes
export default connect(mapStateToProps, mapDispatchToProps)(YourFansPage)
