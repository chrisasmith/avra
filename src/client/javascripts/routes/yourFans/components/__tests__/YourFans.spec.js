import React from 'react'
import { shallow } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { shallowToJson } from 'enzyme-to-json'
import YourFans from '../YourFans'

const mockStore = configureMockStore([thunk])

let store

const getHome = s => shallow(<YourFans store={mockStore(s)} />).shallow()

describe('Home Page', () => {
  beforeEach(() => {
    store = {
      location: { },
      loading: {

      },
      error: {

      },
      filters: {
        filter: {},
      },
      pages: {
        yourFans: {
          data: null,
        },
      },
    }
  })

  it('should show loading', () => {
    store.loading.GET_HOME_PAGE = true

    const wrapper = getHome(store)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should show error', () => {
    store.error.GET_HOME_PAGE = 'bad stuff happened'

    const wrapper = getHome(store)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
