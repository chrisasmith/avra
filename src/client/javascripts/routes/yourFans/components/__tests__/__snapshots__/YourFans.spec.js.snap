// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should show error 1`] = `
<div
  className="columns"
>
  <div>
    <Connect(FilterBar)
      onFilterChange={[Function]}
    />
  </div>
  <div>
    <div
      className="title"
    >
      <h4>
        Your fans
      </h4>
    </div>
    <div
      className="tabs"
    >
      <Tabs
        location={Object {}}
      />
    </div>
  </div>
</div>
`;

exports[`Home Page should show loading 1`] = `
<div
  className="columns"
>
  <div>
    <Connect(FilterBar)
      onFilterChange={[Function]}
    />
  </div>
  <div>
    <div
      className="title"
    >
      <h4>
        Your fans
      </h4>
    </div>
    <div
      className="tabs"
    >
      <Tabs
        location={Object {}}
      />
    </div>
  </div>
</div>
`;
